### PR TITLE
IA-2781 IA-2783: adding phone number to potential payment and payments lots

### PR DESCRIPTION
**Pull request by beygorghor**
Display user phone number on potential payments page and payments views

Related JIRA tickets : IA-2781 IA-2783

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [x] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

- added a column to potential payment table and payments table
- fix sort of those on dialogs

## How to test

Make sure you have a phone number on the users that have potential payments and payments.

Visit potential payment and payments lots pages, make sur phone numbers are present (also in the xlsx).
Try to order data by phone number

## Print screen / video
Screenshot 2024-04-02 at 14 20 54
Screenshot 2024-04-02 at 14 20 22

https://github.com/BLSQ/iaso/assets/12494624/a773ee0a-7a2b-4b54-aab7-5b4dc4a1b92f

Screenshot 2024-04-02 at 14 19 12

### CODE REVIEW

File: iaso/api/payments/views.py [hunk 1]
```diff
@@ -373,13 +381,16 @@ class PotentialPaymentsViewSet(ModelViewSet, AuditMixin):
...
     def get_queryset(self):
-        return (
+        queryset = (
             PotentialPayment.objects.prefetch_related("change_requests")
             .filter(change_requests__created_by__iaso_profile__account=self.request.user.iaso_profile.account)
             # Filter out potential payments already linked to a payment lot as this means there's already a task running converting them into Payment
             .filter(payment_lot__isnull=True)
             .distinct()
         )
+        queryset = queryset.annotate(change_requests_count=Count("change_requests"))
```
Comment (kemar) [inline iaso/api/payments/views.py:391]: It seems to me that the missing annotation should've been detected in unit tests.

File: iaso/api/payments/views.py [hunk 2]
```diff
@@ -264,12 +264,17 @@ def _get_table_row(self, payment, row_num=None, export_type="csv"):
...
             str(payment.id),
             payment.status,
-            str(payment.user.id),  # Added User ID
+            str(payment.user.id),
             payment.user.username,
+            (
+                payment.user.iaso_profile.phone_number.as_e164
+                if payment.user.iaso_profile and payment.user.iaso_profile.phone_number
+                else None
+            ),
```
Comment (kemar) [inline iaso/api/payments/views.py:273]: I think that adding this should've broken something in unit tests.

File: hat/assets/js/apps/Iaso/hooks/useSortedItems.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            columnAccessor
+                ? orderBy(
+                      items,
+                      [item => get(item, columnAccessor)],
+                      [sortOrder],
+                  )
+                : items,
+        [items, columnAccessor, sortOrder],
+    );
+};
```
Comment (kemar) [inline hat/assets/js/apps/Iaso/hooks/useSortedItems.ts:24]: Why are you using `useMemo()` so much here?

The memory cost may outweigh the benefits and React's rendering mechanism is pretty efficient by default.

**Review by kemar — APPROVED** [review]
LGTM (I added some comments but nothing is blocking)

File: hat/assets/js/apps/Iaso/hooks/useSortedItems.ts (hunk 3 above)
Comment (beygorghor) [reply hat/assets/js/apps/Iaso/hooks/useSortedItems.ts:24]: yes you are right

File: iaso/api/payments/views.py (hunk 2 above)
Comment (beygorghor) [reply iaso/api/payments/views.py:273]: yes me too, tests are not good enough on this

**Pull request merged**